src/neapolitan/templatetags/neapolitan.py

```diff
@@ -8,10 +8,14 @@
 
 def action_links(view, object):
     actions = [
-        (Role.DETAIL.reverse(view, object), "View"),
-        (Role.UPDATE.reverse(view, object), "Edit"),
-        (Role.DELETE.reverse(view, object), "Delete"),
+        (url, name)
+        for url, name in [
+            (Role.DETAIL.reverse(view, object), "View"),
+            (Role.UPDATE.reverse(view, object), "Edit"),
+            (Role.DELETE.reverse(view, object), "Delete"),
+        ] if url is not None
```

Comment:
Is this sufficient? 🤔

If I do `BookmarksCRUDView.get_urls(roles={Role.LIST})` will not these not raise a reverse match error? 

Could you add a test case for that example, asserting that the `bookmark-list` view renders correctly, and that these links are absent? (That should cover both the `get_context_data()` case and this one here.)

Reply:
They will not raise an error because of [this change](https://github.com/carltongibson/neapolitan/pull/47/files#diff-15775856a482440af9196b06d3627395adc2f09fae171621cb2cbddb08fa3463R91) which catches the `NoReverseMatch` exception and returns `None` instead.

I'll add a test.

Something I did on other projects was to have an extendable test case that tests if all possible actions render with a satisfactory response code (>=200,<400). But I think that should probably wait until ROLE protocol is a thing

Reply:
> They will not raise an error because of [this change](https://github.com/carltongibson/neapolitan/pull/47/files#diff-15775856a482440af9196b06d3627395adc2f09fae171621cb2cbddb08fa3463R91)... 

Ah, yes. I didn't spot that. Let me consider this. 🤔 (Is it changing the contract of `reverse` too much — often you're going to **want** `NoReverseMatch` when you call the reverse for a non-routed handler )

> I'll add a test.

Great. Thanks. 

> Something I did on other projects ...

Yes! A CRUDViewTestCase kind of thing is on the radar to give base coverage. (But yes, 😜, pending resolving the custom roles issue.)